include/geode/mesh/core/regular_grid.h

```diff
@@ -93,6 +93,20 @@ namespace geode
         absl::optional< Index > previous_cell(
             const Index& index, index_t direction ) const;
 
+        index_t nb_vertices() const;
+
+        index_t nb_vertices( index_t direction ) const;
+
+        Index vertex_index( index_t index ) const;

```

Comment:
Index represents both VertexIndex and CellIndex. Should we explicily separate them ? Maybe with two structs ? This will be a BREAKING change..

@panquez what do you think ?

Reply:
True, but any CellIndex is the same as the VertexIndex of its corner, so I think it also makes sense to keep the same struct for the two (as was already the case with the 'Point<dim> point(const Index& index) const' function).

Reply:
Yes but it does not refer to the same concept. Even if their implementations are equal. I think this is the same than PolygonVertex and PolygonEdge.

Reply:
I agree with this last proposal from Arnaud. IMo CellIndex and VertexIndex should be different